projects/components/src/quick-search/quick-search.provider.ts

```diff
@@ -29,6 +29,13 @@ export interface QuickSearchProvider {
      */
     data: unknown;
 
+    /**
+     * Optional flag to control the visibility of the corresponding section.
```

Comment:
Consider removing `Optional flag` since `?` and `boolean` already imply that.

```
Whether a provider section should be hidden if there are no results.
 * When `true` the section will not be displayed if it contains no data.
 * Defaults to `false` and in that case the section is always visible, rendering 'No results found.'
```